### PR TITLE
ASM-6253 Configring spanning-tree-protocol on switch

### PR DESCRIPTION
**Pull request by prem548**
Code changes will configure Spanning-tree-protocol on switch based on user selection.

### CODE REVIEW

File: lib/puppet/provider/force10_settings/dell_ftos.rb [hunk 1]
```diff
@@ -65,5 +65,44 @@ def send_cmd(command, context='')
...
+    update_protocol(session, spanning_tree_protocol, existing_protocol)
+  end
+
+  def show_protocol(session)
+    meta_data = session.command('show running-config') || ''
+    result = []
+    match_data = /protocol spanning\-tree\s+(\w+)/
+    meta_data.split("\n").each do |line|
+      stp = line.match(match_data)
+      if !stp.nil?
```
Comment (ripienaar) [inline lib/puppet/provider/force10_settings/dell_ftos.rb]: this can just be `result << stp[1] if stp`

Comment (prem548) [inline lib/puppet/provider/force10_settings/dell_ftos.rb]: possibly, I will update that loop

File: lib/puppet/provider/force10_settings/dell_ftos.rb [hunk 2]
```diff
@@ -65,5 +65,45 @@ def send_cmd(command, context='')
...
+      stp = line.match(match_data)
+      result << stp[1] if stp
+    end
+    result
+  end
+
+  def update_protocol(session,protocol,existing_protocol)
+    #Since we are receiving RSTP, PVST
+    protocol = protocol.downcase
+    if protocol.match(/one/)
```
Comment (gavin-scott) [inline lib/puppet/provider/force10_settings/dell_ftos.rb]: Is this matching "none"?

File: lib/puppet/provider/force10_settings/dell_ftos.rb [hunk 3]
```diff
@@ -65,5 +65,45 @@ def send_cmd(command, context='')
...
+    disable_stp.each do |stp|
+      session.command("no protocol spanning-tree #{stp}")
+    end
+    enable_stp = protocol - existing_protocol
+    enable_stp.each do |stp|
+      session.command("protocol spanning-tree #{stp}")
+      session.command("no disable")
+    end
+    session.command("end")
+  end
```
Comment (gavin-scott) [inline lib/puppet/provider/force10_settings/dell_ftos.rb]: Can you add some unit tests for these methods?

File: lib/puppet/provider/force10_settings/dell_ftos.rb (hunk 2 above)
Comment (prem548) [inline lib/puppet/provider/force10_settings/dell_ftos.rb]: None from UI or none

Comment (gavin-scott) [inline lib/puppet/provider/force10_settings/dell_ftos.rb]: How about `protocol.downcase == "none"` then? That would read a little easier.

Comment (prem548) [inline lib/puppet/provider/force10_settings/dell_ftos.rb]: I will update the changes, Make new PR with test cases

Comment (gavin-scott) [inline lib/puppet/provider/force10_settings/dell_ftos.rb]: It's better to keep the same PR and just update this one.

Comment (prem548) [inline lib/puppet/provider/force10_settings/dell_ftos.rb]: I apologize, I meant I will amend  changes to this commit

**Comment by prem548** [comment]
@gavin-scott  Could you give glance on changes I made on this PR. Please let me know, If any changes required.

File: spec/unit/puppet/provider/force10_settings/dell_ftos_spec.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+#provider_path = Pathname.new(__FILE__).parent.parent
```
Comment (gavin-scott) [inline spec/unit/puppet/provider/force10_settings/dell_ftos_spec.rb]: Delete this

File: spec/unit/puppet/provider/force10_settings/dell_ftos_spec.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+require 'spec_helper'
+require 'rspec/expectations'
+require 'json'
+require 'puppet'
+require 'puppet_x/force10/transport'
+provider_class  = Puppet::Type.type(:force10_settings).provider(:dell_ftos)
```
Comment (gavin-scott) [inline spec/unit/puppet/provider/force10_settings/dell_ftos_spec.rb]: space before and after this line

File: spec/unit/puppet/provider/force10_settings/dell_ftos_spec.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+describe provider_class do
+  let(:resource) {Puppet::Type.type(:force10_settings).new(
+                                                         {
+                                                              :name => 'dell_iom-172.17.2.178',
+                                                              :hostname => "DELL",
+                                                              :spanning_tree_mode => "PVST"
+                                                         }
+  )}
+   let(:provider) {resource.provider}
```
Comment (gavin-scott) [inline spec/unit/puppet/provider/force10_settings/dell_ftos_spec.rb]: Indentation is off, has 3 spaces instead of 2.

**Comment by gavin-scott** [comment]
Minor formatting comments, otherwise looks good. Ping me when the formatting changes are in.

**Pull request merged**